### PR TITLE
API 1.34 - Keem Bay platform

### PR DESCRIPTION
**Pull request by onabiull**
Incorporate changes from 1.34 published in [oneVPL spec](https://spec.oneapi.com/versions/latest/elements/oneVPL/source/index.html).

### CODE REVIEW

File: api/include/mfxstructures.h [hunk 1]
```diff
@@ -965,6 +973,12 @@ enum {
 #if (MFX_VERSION >= 1031)
     MFX_EXTBUFF_PARTIAL_BITSTREAM_PARAM         = MFX_MAKEFOURCC('P','B','O','P'),
 #endif
+
+#if (MFX_VERSION >= 1034)
+    MFX_EXTBUFF_ENCODER_IPCM_AREA  = MFX_MAKEFOURCC('P', 'C', 'M', 'R'),
+    MFX_EXTBUFF_INSERT_HEADERS  = MFX_MAKEFOURCC('S', 'P', 'R', 'E'),
```
Comment (mgonchar) [inline api/include/mfxstructures.h:979]: return alignment please

File: api/include/mfxstructures.h [hunk 2]
```diff
@@ -1675,9 +1690,21 @@ MFX_PACK_END()
 /* MBQPMode */
 enum {
     MFX_MBQP_MODE_QP_VALUE = 0, // supported in CQP mode only
-    MFX_MBQP_MODE_QP_DELTA = 1
+    MFX_MBQP_MODE_QP_DELTA = 1,
+	MFX_MBQP_MODE_QP_ADAPTIVE = 2
```
Comment (mgonchar) [inline api/include/mfxstructures.h:1694]: some extra spaces in beginning

File: api/include/mfxstructures.h [hunk 3]
```diff
@@ -1675,9 +1690,21 @@ MFX_PACK_END()
...
 
+#if (MFX_VERSION >= 1034)
+MFX_PACK_BEGIN_USUAL_STRUCT()
+typedef struct{
+    union {
+        mfxU8 QP;
+        mfxI8 DeltaQP;
+    };
+    mfxU16 Mode;
+} mfxQPandMode;
```
Comment (mgonchar) [inline api/include/mfxstructures.h:1705]: should appear in file with static asserts

File: api/include/mfxstructures.h [hunk 4]
```diff
@@ -1689,11 +1716,41 @@ typedef struct {
...
 MFX_PACK_END()
 
+#if (MFX_VERSION >= 1034)
+MFX_PACK_BEGIN_USUAL_STRUCT()
+typedef struct {
+    mfxExtBuffer    Header; /* Extension buffer header. Header.BufferId must be equal to MFX_EXTBUFF_INSERT_HEADERS. */
+    mfxU16          SPS;      /* tri-state option to insert SPS */
+    mfxU16          PPS;      /* tri-state option to insert PPS */
+    mfxU16          reserved[8];
+} mfxExtInsertHeaders;
```
Comment (mgonchar) [inline api/include/mfxstructures.h:1734]: safe for asserts

File: api/include/mfxstructures.h [hunk 5]
```diff
@@ -1689,11 +1716,41 @@ typedef struct {
...
+    mfxU16          NumArea;  /* Number of Area's */
+    struct {
+        mfxU32      Left; /* Left Area's coordinate. */
+        mfxU32      Top;  /* Top  Area's coordinate. */
+        mfxU32      Right; /* Right Area's coordinate. */
+        mfxU32      Bottom; /* Bottom Area's coordinate. */
+
+        mfxU16      reserved2[8];
+    } Area[64];
+} mfxExtEncoderIPCMArea;
```
Comment (mgonchar) [inline api/include/mfxstructures.h:1751]: same

File: api/include/mfxstructures.h [hunk 6]
```diff
@@ -1689,11 +1716,41 @@ typedef struct {
...
+
+    mfxU16          NumArea;  /* Number of Area's */
+    struct {
+        mfxU32      Left; /* Left Area's coordinate. */
+        mfxU32      Top;  /* Top  Area's coordinate. */
+        mfxU32      Right; /* Right Area's coordinate. */
+        mfxU32      Bottom; /* Bottom Area's coordinate. */
+
+        mfxU16      reserved2[8];
+    } Area[64];
```
Comment (mgonchar) [inline api/include/mfxstructures.h:1750]: why not dynamic array?

Comment (onabiull) [reply api/include/mfxstructures.h:1750]: I'd suggest to ask in [oneVPL spec](https://spec.oneapi.com/versions/latest/elements/oneVPL/source/VPL_intel_media_sdk.html), @varistar would you comment?

File: doc/mediasdk-man.md [hunk 7]
```diff
@@ -9146,6 +9334,8 @@ SDK API 1.27 adds `MFX_PLATFORM_ICELAKE`.
 
 SDK API 1.31 adds `MFX_PLATFORM_ELKHARTLAKE`, `MFX_PLATFORM_JASPERLAKE`, `MFX_PLATFORM_TIGERLAKE`.
 
+SDK API 1.34 adds `MFX_PLATFORM_KEEMBAY`.
```
Comment (dvrogozh) [inline doc/mediasdk-man.md:9337]: I don't see `MFX_PLATFORM_KEEMBAY` in the API headers .h files! Did you forget to add?

**Review by dvrogozh — DISMISSED** [review]

File: doc/mediasdk-man.md (hunk 7 above)
Comment (varistar) [reply doc/mediasdk-man.md:9337]: Good point, yes, need to add.

File: api/include/mfxstructures.h (hunk 6 above)
Comment (varistar) [reply api/include/mfxstructures.h:1750]: Yep, makes sense to add dynamic allocation especially if we have Num above.

File: _studio/shared/src/mfx_static_assert_structs.cpp [hunk 8]
```diff
@@ -350,6 +350,9 @@
         MSDK_STATIC_ASSERT_STRUCT_SIZE(mfxExtDecVideoProcessing     ,132  )
         MSDK_STATIC_ASSERT_STRUCT_SIZE(mfxExtChromaLocInfo          ,32   )
         MSDK_STATIC_ASSERT_STRUCT_SIZE(mfxExtMBQP                   ,64   )
+        MSDK_STATIC_ASSERT_STRUCT_SIZE(mfxQPandMode                 ,4    )
```
Comment (mgonchar) [inline _studio/shared/src/mfx_static_assert_structs.cpp:353]: you also have to specify offsets of each field

File: _studio/shared/src/mfx_static_assert_structs.cpp [hunk 9]
```diff
@@ -350,6 +350,10 @@
...
```
Comment (mgonchar) [inline _studio/shared/src/mfx_static_assert_structs.cpp:353]: ```MFX_VERSION``` guard is missing

File: _studio/shared/src/mfx_static_assert_structs.cpp [hunk 10]
```diff
@@ -1362,6 +1366,24 @@
         MSDK_STATIC_ASSERT_STRUCT_OFFSET(mfxExtMBQP                         ,QP                            ,56   )
         MSDK_STATIC_ASSERT_STRUCT_OFFSET(mfxExtMBQP                         ,DeltaQP                       ,56   )
 
+
+        MSDK_STATIC_ASSERT_STRUCT_OFFSET(mfxQPandMode                       ,QP                            ,0    )
```
Comment (mgonchar) [inline _studio/shared/src/mfx_static_assert_structs.cpp:1370]: same here

File: api/include/mfxstructures.h [hunk 11]
```diff
@@ -1689,11 +1716,41 @@ typedef struct {
     union {
         mfxU8  *QP;         // Block QP value. Valid when Mode = MFX_MBQP_MODE_QP_VALUE
         mfxI8  *DeltaQP;    // For block i: QP[i] = BrcQP[i] + DeltaQP[i]. Valid when Mode = MFX_MBQP_MODE_QP_DELTA
+#if (MFX_VERSION >= 1034)
+        mfxQPandMode *QPmode; // Block-granularity modes when MFX_MBQP_MODE_QP_ADAPTIVE is set
```
Comment (mgonchar) [inline api/include/mfxstructures.h:1720]: I know that this is union, but still we have both ```QP``` and ```DeltaQP``` in static asserts file, let's also add ```QPmode```

**Review by mgonchar — DISMISSED** [review]

File: _studio/shared/src/mfx_static_assert_structs.cpp (hunk 9 above)
Comment (onabiull) [reply _studio/shared/src/mfx_static_assert_structs.cpp:353]: I'm going to get rid of them soon.

File: api/include/mfxstructures.h (hunk 11 above)
Comment (onabiull) [reply api/include/mfxstructures.h:1720]: ok

File: doc/mediasdk-man.md (hunk 7 above)
Comment (onabiull) [reply doc/mediasdk-man.md:9336]: done

File: _studio/shared/src/mfx_static_assert_structs.cpp (hunk 9 above)
Comment (mgonchar) [reply _studio/shared/src/mfx_static_assert_structs.cpp:353]: But you have such guards introduced by this commit. I support your intention to remove them, but for now I'm just asking for consistency.

**Review by alariono — APPROVED** [review]

**Pull request merged**